Library/Homebrew/test/download_strategies_spec.rb

```diff
@@ -0,0 +1,228 @@
+require "download_strategy"
+
+describe AbstractDownloadStrategy do
+  subject { described_class.new(name, resource) }
+  let(:name) { "foo" }
+  let(:resource) { double(Resource, url: "http://example.com/foo.tar.gz", mirrors: [], specs: {}, version: nil) }
+  let(:args) { %w[foo bar baz] }
+
+  describe "#expand_safe_system_args" do
+    it "works with an explicit quiet flag" do
+      args << { quiet_flag: "--flag" }
+      expanded_args = subject.expand_safe_system_args(args)
+      expect(expanded_args).to eq(%w[foo bar baz --flag])
+    end
+
+    it "adds an implicit quiet flag" do
+      expanded_args = subject.expand_safe_system_args(args)
+      expect(expanded_args).to eq(%w[foo bar -q baz])
+    end
+
+    it "does not mutate the arguments" do
+      result = subject.expand_safe_system_args(args)
+      expect(args).to eq(%w[foo bar baz])
+      expect(result).not_to be args
```

Comment:
I know this is how it was before, but if you feel like it, I think this could use a more thorough test that actually tries mutating the arguments after passing them.